presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -1102,10 +1110,25 @@ public MetastoreOperationResult addConstraint(MetastoreContext metastoreContext,
 
     public List<TableConstraint<String>> getTableConstraints(MetastoreContext metastoreContext, String schemaName, String tableName)
     {
-        return readConstraintsFile(schemaName, tableName).stream()
+        Set<TableConstraint> rawConstraints = readConstraintsFile(schemaName, tableName);
+
+        List<TableConstraint<String>> constraints = rawConstraints.stream()
```

Comment:
Instead of streaming 3 times, how about we just forEach over the rawConstraints and check and add the constraints to a list builder? Or is it a hard requirement that we add the PK, then unique and then the Not Null constraints ?

Reply:
It leads to non-determinism in the tests if these are not in order. Since this is the file-based metastore anyway I am not too concerned about the performance overhead